build/cmake/functions.cmake

```diff
@@ -294,16 +300,19 @@ function(wx_set_target_properties target_name is_base)
             )
     endif()
 
+    file(RELATIVE_PATH wxSETUP_HEADER_REL ${wxOUTPUT_DIR} ${wxSETUP_HEADER_PATH})
     target_include_directories(${target_name}
         BEFORE
         PUBLIC
-            ${wxSETUP_HEADER_PATH}
-            ${wxSOURCE_DIR}/include
+            $<BUILD_INTERFACE:${wxSETUP_HEADER_PATH}>
+            $<BUILD_INTERFACE:${wxSOURCE_DIR}/include>
+            $<INSTALL_INTERFACE:lib/${wxSETUP_HEADER_REL}>
+            $<INSTALL_INTERFACE:include>
         )
 
-    if(wxTOOLKIT_INCLUDE_DIRS AND NOT is_base)
+    if(wxTOOLKIT_INCLUDE_DIRS AND NOT wxTARGET_IS_BASE)
         target_include_directories(${target_name}
-            PUBLIC ${wxTOOLKIT_INCLUDE_DIRS})
+            PRIVATE ${wxTOOLKIT_INCLUDE_DIRS})
```

Comment:
I have no idea with this part. Other guys should review this part again.

Reply:
There is no need to expose all the stuff used to build wxWidgets. Projects using it don't need to know that wxWidgets includes some gtk or libsecret or gstreamer or webkit headers. The only thing they need is the wxWidgets headers.